src/components/PopupWithForm.js

```diff
@@ -2,32 +2,35 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super({ popupSelector });
+    super(popupSelector);
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
-  }
-
-  open() {
-    super.open();
+    this._submitBtn = this._popupElement.querySelector(".modal__button");
+    this._submitBtnText = this._submitBtn.textContent;
   }
 
   _getInputValues() {
-    const inputs = this._popupElement.querySelectorAll(".modal__input");
-    const inputsItems = {};
+    const inputs = this._popupForm.querySelectorAll(".modal__input");
+    const inputItems = {};
 
-    inputs.forEach(({ name, value }) => {
-      inputsItems[name] = value;
+    inputs.forEach((input) => {
+      inputItems[input.name] = input.value;
     });
-    return inputsItems;
+    return inputItems;
+  }
+  renderLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitBtn.textContent = loadingText;
+    } else {
+      this._submitBtn.textContent = this._submitBtnText;
+    }
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`